flink-tests/src/test/java/org/apache/flink/test/checkpointing/RegionFailoverITCase.java

```diff
@@ -102,8 +111,14 @@
 
 	@Before
 	public void setup() throws Exception {
+		HighAvailabilityServicesUtilsTest.TestHAFactory.haServices = new TestingHaServices(
```

Comment:
please don't re-use the TestHAFactory class; with it's static mutable field it should be avoided if possible.
You can just define your own `HighAvailabilityServicesFactory` with a lambda
`HighAvailabilityServicesFactory haServiceFactory = (config, executor) -> new TestingHaServices...`